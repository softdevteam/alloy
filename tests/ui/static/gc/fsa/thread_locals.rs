#![feature(gc)]
#![feature(negative_impls)]
#![allow(dead_code)]
#![allow(unused_variables)]
include!{"./auxiliary/types.rs"}

use std::cell::Cell;

thread_local! {
    static COUNTER: Cell<u32> = Cell::new(0);
}


#[derive(Debug)]
struct S;

impl Drop for S {
    fn drop(&mut self) {
        // Access the thread-local variable
        let x = COUNTER.get();
    }
}

fn main() {
    Gc::new(FinalizerUnsafeWrapper(S));
    //~^ ERROR: The drop method for `S` cannot be safely finalized.
}
