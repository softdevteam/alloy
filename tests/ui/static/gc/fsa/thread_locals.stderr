error: The drop method for `S` cannot be safely finalized.
  --> $DIR/thread_locals.rs:25:13
   |
LL |         let x = COUNTER.get();
   |                 ------- this thread-local is not safe to run in a finalizer
...
LL |     Gc::new(FinalizerUnsafeWrapper(S));
   |     --------^^^^^^^^^^^^^^^^^^^^^^^^^- caused by trying to construct a `Gc<FinalizerUnsafeWrapper<S>>` here.
   |
   = help: `Gc` runs finalizers on a separate thread, so thread-locals cannot be accessed.

error: aborting due to 1 previous error

